### PR TITLE
IgxTreeGrid - Editing

### LINKED ISSUE #2908: IgxTreeGrid - Add row editing + transactions to tree grid
### Description  
Integrate row editing and transactions in the tree grid

 * igniteui-angular version:  6.2.x
 * browser: N/A

### PR DESCRIPTION
**Pull request by ViktorSlavov**
Closes #2908.

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts [hunk 1]
```diff
@@ -366,6 +366,277 @@ describe('IgxTreeGrid - Expanding / Collapsing', () => {
...
+                IgxTreeGridModule]
+        })
+        .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fix = TestBed.createComponent(IgxTreeGridRowEditingComponent);
+        fix.detectChanges();
+        treeGrid = fix.componentInstance.treeGrid;
+        fix.detectChanges();
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts]: The last detectChanges() should not be necessary.

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts [hunk 2]
```diff
@@ -366,6 +366,277 @@ describe('IgxTreeGrid - Expanding / Collapsing', () => {
...
+        const cell = treeGrid.getCellByColumn(1, 'Name');
+        cell.inEditMode = true;
+        tick();
+        fix.detectChanges();
+
+        const firstRow = rows[0];
+        const indicatorDiv = TreeGridFunctions.getExpansionIndicatorDiv(firstRow);
+        indicatorDiv.triggerEventHandler('click', new Event('click'));
+        tick();
+        fix.detectChanges();
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts]: Are tick() + detectChanges() necessary? triggerEventHandler should not require those.

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts [hunk 3]
```diff
@@ -366,6 +366,277 @@ describe('IgxTreeGrid - Expanding / Collapsing', () => {
...
+
+        treeGrid.toggleRow(treeGrid.getRowByIndex(0).rowID);
+        tick();
+        fix.detectChanges();
+        expect(treeGrid.rowEditingOverlay.collapsed).toBeTruthy();
+
+        treeGrid.toggleRow(treeGrid.getRowByIndex(0).rowID);
+        tick();
+        fix.detectChanges();
+        expect(treeGrid.rowEditingOverlay.collapsed).toBeTruthy();
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-expanding.spec.ts]: Should it not be expanded at least once in this test?

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts [hunk 4]
```diff
@@ -277,4 +287,231 @@ describe('IgxTreeGrid - Integration', () => {
             expect(parseInt(column.width, 10)).toBe(152);
         });
     });
+
+    describe('Row editing', () => {
+        beforeEach(() => {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts]: Add configureTestSuite().

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts [hunk 5]
```diff
@@ -277,4 +287,231 @@ describe('IgxTreeGrid - Integration', () => {
...
+            // The banner appears below the row
+            expect(bannerTop).toBeGreaterThanOrEqual(editRowBottom);
+            // No much space between the row and the banner
+            expect(bannerTop - editRowBottom).toBeLessThan(2);
+        }));
+
+        xit('shows the banner above the edited parent node if it is the last one', fakeAsync(() => {
+            const grid = fix.componentInstance.treeGrid as IgxTreeGridComponent;
+            grid.height = '200px';
+            fix.detectChanges();
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts]: From my experience, setting grid.height seems to require await or at least a tick().

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts [hunk 6]
```diff
@@ -277,4 +287,231 @@ describe('IgxTreeGrid - Integration', () => {
...
+            cell.inEditMode = true;
+            tick();
+            fix.detectChanges();
+
+            const editRow = cell.row.nativeElement;
+            const banner = fix.debugElement.query(By.css('.' + CSS_CLASS_BANNER)).nativeElement;
+
+            const bannerBottom = banner.getBoundingClientRect().bottom;
+            const editRowTop = editRow.getBoundingClientRect().top;
+            console.log(banner.getBoundingClientRect());
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts]: I guess these console.logs are left here by accident.

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts [hunk 7]
```diff
@@ -277,4 +287,231 @@ describe('IgxTreeGrid - Integration', () => {
...
+            const bannerBottom = banner.getBoundingClientRect().bottom;
+            const editRowTop = editRow.getBoundingClientRect().top;
+
+            // The banner appears below the row
+            expect(bannerBottom).toBeLessThanOrEqual(editRowTop);
+            // No much space between the row and the banner
+            expect(editRowTop - bannerBottom).toBeLessThan(2);
+        }));
+
+        fit('banner hides when you expand/collapse the edited row', fakeAsync(() => {
```
Comment (gedinakova) [inline projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts]: Remove the focus from this test.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 8]
```diff
@@ -900,12 +900,18 @@
...
+            color: --var($theme, 'cell-edited-value-color');
+        }
     }
 
     %igx-grid__tr--deleted {
-        text-decoration-line: line-through;
+        %grid-cell-text {
+            font-style: italic;
+            color: igx-color(map-get($theme, 'palette'), 'error');
+            text-decoration-line: line-through;
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: @simeonoff Can't this be just `text-decoration` so it'd work in IE/Edge too?
@StefanIvanov Is red part of the design?

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: @damyanpetev text-decoration-line is just a property of the shorthand text-decoration. The `line-through` text decoration is not supported in IE and Edge. As to the color being red, I've discussed it wit @StefanIvanov before implementing the change.

Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss]: @simeonoff Despite being the shorthand, `text-decoration: line-through;` is in the CSS1 spec and [works just fine even in older IE-s](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration#Browser_compatibility). So unless we need any of the decorations, can we use the shorthand for consistency?

File: projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts (hunk 4 above)
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-integration.spec.ts]: Not needed unless the test suite calls `TestBed.configureTestingModule(/*...*/).compileComponents();`

**Pull request merged**